data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -0,0 +1,180 @@
+                            when {
+                                !transactionDto.transfers.isNullOrEmpty() -> addAll(transactionDto.transfers.map { transfer(it) })
+                                transactionDto.transfers.isNullOrEmpty() && !transactionDto.data.hexStringNullOrEmpty() -> add(custom(transactionDto))
+                                else -> add(transfer(transactionDto))
+                            }
+                        }
+                        is MultisigTransactionDto -> {
+                            when {
+                                isSettingsChange(transactionDto) -> add(settings(transactionDto))
+                                isErc20Transfer(transactionDto) -> add(transferErc20(transactionDto))
+                                isErc721Transfer(transactionDto) -> add(transferErc721(transactionDto))
+                                isEthTransfer(transactionDto) -> add(transferEth(transactionDto))
+                                else -> add(custom(transactionDto))
+                            }
+                        }
+                        else -> add(custom(transactionDto))
+                    }
+                }
+            }
+
+    private fun isErc721Transfer(transactionDto: MultisigTransactionDto): Boolean =
+        transactionDto.operation == Operation.CALL &&
+                transactionDto.contractInfo?.type == ContractInfoType.ERC721 && // Always false unless wir have contractInfo
+                listOf("safeTransferFrom", "transferFrom").contains(transactionDto.dataDecoded?.method)
+
+    private fun isErc20Transfer(transactionDto: MultisigTransactionDto): Boolean =
+        transactionDto.operation == Operation.CALL &&
+//                transactionDto.contractInfo?.type == ContractInfoType.ERC20 && //TODO enable this check when we have contractInfo
+                listOf("transfer", "transferFrom").contains(transactionDto.dataDecoded?.method)
+
+    private fun isEthTransfer(transactionDto: MultisigTransactionDto): Boolean =
+        transactionDto.data == null && transactionDto.operation == Operation.CALL
+
+    private fun isSettingsChange(transactionDto: MultisigTransactionDto): Boolean =
+        transactionDto.to == transactionDto.safe
+                && transactionDto.operation == Operation.CALL
+                && SafeRepository.isSettingsMethod(transactionDto.dataDecoded?.method)
+
+    private fun transfer(transferDto: TransferDto): Transaction.Transfer {
+        return Transaction.Transfer(
+            transferDto.to,
+            transferDto.from,
+            transferDto.value,
+            transferDto.executionDate?.formatBackendDate(),
+            transferDto.type.let {
+                when (it) {
+                    TransferType.ERC20_TRANSFER -> TokenRepository.FAKE_ERC20_TOKEN_INFO
+                    TransferType.ERC721_TRANSFER -> TokenRepository.FAKE_ERC721_TOKEN_INFO
+                    else -> TokenRepository.ETH_SERVICE_TOKEN_INFO
```

Comment:
ah, ok. I see why you had to leave those mock values outside of tests. If you don't mind, I would prefer moving those over to this class, so we have all the stuff that needs removing or mocking in the same place, that is the `TransactionRepository` and then when the backend is ready we just check one place.